index.bs

```diff
@@ -1947,6 +1973,111 @@ This
+        <code>|options|.{{CredentialRequestOptions/publicKey}}</code> member contains a {{PublicKeyCredentialRequestOptions}}
+        object specifying the desired attributes of the [=public key credential=] to discover.
+</dl>
+
+The steps for [=issuing a credential request to an authenticator=] are as follows:
+
+    1. If <code>|options|.{{PublicKeyCredentialRequestOptions/userVerification}}</code> is set to
+        {{UserVerificationRequirement/required}} and the |authenticator| is not capable of performing [=user verification=],
+        [=iteration/continue=].
+
+    1. Let |userVerification| be the <dfn>effective user verification requirement for assertion</dfn>, a Boolean value, as
+        follows. If <code>|options|.{{PublicKeyCredentialRequestOptions/userVerification}}</code>
+
+            <dl class="switch">
+
+                :   is set to {{UserVerificationRequirement/required}}
+                ::  Let |userVerification| be [TRUE].
+
+                :   is set to {{UserVerificationRequirement/preferred}}
+                ::  If the |authenticator|
+
+                    <dl class="switch">
+                        :   is capable of [=user verification=]
+                        ::  Let |userVerification| be [TRUE].
+
+                        :   is not capable of [=user verification=]
+                        ::  Let |userVerification| be [FALSE].
+                    </dl>
+
+                :   is set to {{UserVerificationRequirement/discouraged}}
+                ::  Let |userVerification| be [FALSE].
+
+            </dl>
+
+    1. <span id="allowCredentialDescriptorListCreation"></span>
+        If <code>|options|.{{PublicKeyCredentialRequestOptions/allowCredentials}}</code>
+        <dl class="switch">
+            :   [=list/is not empty=]
+            ::  1. Let |allowCredentialDescriptorList| be a new [=list=].
+
+                1. Execute a [=client platform=]-specific procedure to determine which, if any, [=public key credentials=] described by
+                    <code>|options|.{{PublicKeyCredentialRequestOptions/allowCredentials}}</code> are [=bound credential|bound=] to this
+                    |authenticator|, by matching with |rpId|,
+                    <code>|options|.{{PublicKeyCredentialRequestOptions/allowCredentials}}.{{PublicKeyCredentialDescriptor/id}}</code>,
+                    and
+                    <code>|options|.{{PublicKeyCredentialRequestOptions/allowCredentials}}.{{PublicKeyCredentialDescriptor/type}}</code>.
+                    Set |allowCredentialDescriptorList| to this filtered list.
+
+                1. If |allowCredentialDescriptorList| [=list/is empty=], [=continue=].
+
+                1. Let |distinctTransports| be a new [=ordered set=].
+
+                1. If |allowCredentialDescriptorList| has exactly one value, set
+                    <code>|savedCredentialIds|[|authenticator|]</code> to <code>|allowCredentialDescriptorList|[0].id</code>'s
```

Comment:
> maybe we should just state that this is a sub-algorithm rather than a standalone one?

The introductory paragraph to section 5.1.4.1. Issuing a Credential Request to an Authenticator attempts to state that, but without using a term like "sub-algorithm".  It could probably be more clear and perhaps use that latter term.

I also continue to feel that having this "Issuing a Credential Request" section be after the "[[DiscoverFromExternalSource]]" section would be better, [as discussed in a prior comment thread](https://github.com/w3c/webauthn/pull/1576#discussion_r739866766) (and in following comments there). Perhaps this will help convince @nsatragno? ;-)

Reply:
Alright, I have moved this algorithm below [[DiscoverFromExternalSource]], passed |savedCredentialIds| as a parameter to avoid referencing it out of nowhere, and declared it as a subalgorithm of [[DiscoverFromExternalSource]]. PTAL.

Reply:
LGTM, thanks for doing all that.